### PR TITLE
Improve Tab system of the Home Activity

### PR DESCRIPTION
**Pull request by GabrielFleischer**
Improve the tab system so that it is more robust, and it is easier to manually switch tabs.

The main idea is that the change of fragments are done from the navigation bar callback and to change tabs manually, we change the value of a live data that impacts the navigation bar.

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 1]
```diff
@@ -149,97 +144,99 @@ public static HomeViewModel obtainViewModel(FragmentActivity activity) {
...
+      showWalletWarning();
+      return false;
     }
-  }
 
-  public void openConnect() {
-    if (checkSelfPermission(Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
+    // Check for the permission, if it is not granted, ask for it
+    if (checkSelfPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED)
+      requestCameraPermission();
```
Comment (pierluca) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:196]: I'm not sure if this is something that needs to be edited in the `.editorconfig`, but please always have curly brackets {} for if/else clauses. It prevents subtle mistakes.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 2]
```diff
@@ -149,97 +144,99 @@ public static HomeViewModel obtainViewModel(FragmentActivity activity) {
...
+    return true;
   }
 
-  public void requestCameraPermission() {
+  private void requestCameraPermission() {
     // Setup result listener to open the connect tab once the permission is granted
     getSupportFragmentManager()
         .setFragmentResultListener(
-            CameraPermissionFragment.REQUEST_KEY, this, (k, b) -> openConnect());
+            CameraPermissionFragment.REQUEST_KEY, this, (k, b) -> openConnectTab());
```
Comment (pierluca) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:207]: what happens if the permission is not granted?

**Review by pierluca — CHANGES_REQUESTED** [review]
Looks good, barring a minor style change

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:196]: Probably yes, I'll look into that in another PR.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:207]: The `openConnectTab()` function is called, which re-ask for permission if needed.

Ideally, we would just not ask for it again and use the (already in place) button in the scanning fragment.

**Review by pierluca — APPROVED** [review]

**Pull request merged**